nomad/state/state_store.go

```diff
@@ -795,19 +825,40 @@ func (s *StateStore) UpsertNode(index uint64, node *structs.Node) error {
 		return fmt.Errorf("csi plugin update failed: %v", err)
 	}
 
+	return nil
+}
+
+// DeleteNode deregisters a batch of nodes
+func (s *StateStore) DeleteNodeCtx(ctx context.Context, index uint64, nodes []string) error {
```

Comment:
By splitting this method out into one version with `ctx` and the other without, we're not forcing all consumers of this state store function to update and thread their changes through the event store. That's maybe convenient for tests but seems like it could be a source of bugs later.

Reply:
Once these changes land in master, I was thinking DeleteNodeCtx would go away and `DeleteNode` or any other state store func would get a context as its first argument. I refrained from doing so here mainly because of tests, there are 252 invocations for `UpsertNode` in test files and since this PR is going into a long lived feature branch I just honestly didn't want to deal with all the potential merge conflicts :sweat_smile: 

If this route is agreed upon by everyone I'll go back and comment / TODO it being refactored away

Reply:
I totally missed this wasn't going to master, so 👍 on that approach.